pyogrio/tests/test_raw_io.py

```diff
@@ -522,6 +522,72 @@ def test_read_write_null_geometry(tmp_path, ext):
     assert np.array_equal(result_fields[0], field_data[0])
 
 
+def test_write_float_nan_null(tmp_path):
+    geometry = np.array(
+        [bytes.fromhex("010100000000000000000000000000000000000000")] * 2,
+        dtype=object,
+    )
+    field_data = [np.array([1.5, np.nan], dtype="float64")]
```

Comment:
Just to make sure we have it covered, would you mind parametrizing this test with both `float64` and `float32`?